Source/RxSwift/cascade.swift

```diff
@@ -23,96 +23,88 @@ extension Observable where Element : ObservableType {
 	- parameter observables: a sequence of observables which will all be immediately subscribed to
 	- returns: An observable sequence that contains elements from the latest observable sequence that emitted elements
 	*/
-	
-	
-	public static func cascade<S : Sequence>(_ observables : S) -> Observable<T> where S.Iterator.Element == Element {
-		let flow = Array(observables)
-		if flow.isEmpty {
-			return Observable<T>.empty()
-		}
-		
-		return Observable<T>.create { observer in
-			var current = 0, initialized = false
-			var subscriptions: [SerialDisposable?] = flow.map { _ in SerialDisposable() }
-
-			let lock = NSRecursiveLock()
-			lock.lock()
-			defer { lock.unlock() }
-			
-			for i in 0 ..< flow.count {
-				let index = i
-				let disposable = flow[index].subscribe { event in
-					lock.lock()
-					defer { lock.unlock() }
-					
-					switch event {
-					case .next(let element):
-						while current < index {
-							subscriptions[current]?.dispose()
-							subscriptions[current] = nil
-							current += 1
-						}
-						if index == current {
-							assert(subscriptions[index] != nil)
-							observer.onNext(element)
-						}
-						
-					case .completed:
-						if index >= current {
-							if (initialized) {
-								subscriptions[index]?.dispose()
-								subscriptions[index] = nil
-								for next in current ..< subscriptions.count {
-									if subscriptions[next] != nil {
-										return
-									}
-								}
-								observer.onCompleted()
-							}
-						}
-						
-					case .error(let error):
-						observer.onError(error)
-					}
-				}
-				if let serialDisposable = subscriptions[index] {
-					serialDisposable.disposable = disposable
-				} else {
-					disposable.dispose()
-				}
-			}
-
-			initialized = true
-			
-			for i in 0 ..< flow.count {
-				if subscriptions[i] != nil {
-					return Disposables.create {
-						subscriptions.forEach { $0?.dispose() }
-					}
-				}
-			}
-
-			observer.onCompleted()
-			return Disposables.create()
-		}
+	public static func cascade<S: Sequence>(_ observables: S) -> Observable<T> where S.Iterator.Element == Element {
```

Comment:
The big changes here is just spacing.
This seems to be the only file to use Tabs instead of Spaces, so changing one portion of the code made everything look strange, so I just re-indented it.

Reply:
The old tabs-vs-spaces war eh? Always been a tabs guy, never understood the purpose of spaces, as:
- some developers like to have 2, 3, 4 or 8 spaces indentation. Using a tab makes it flexible
- using tabs makes smaller files (yeah I know in this day and age, not important)
:)

Reply:
Well, it's not a war or anything. I just think people mostly use spaces, because it ensures the files look the same wherever you go. 

I don't mind going one way or the other but then we need to see if we go the same everywhere. Otherwise, if you use tabs and someone who uses spaces edits your files, it will look very strange (like happened for me).

Good either way :)

Reply:
I remember that editors like Emacs and VIM support a comment indicator that tells whether a file uses spaces or tabs, and adjust accordingly. AppCode and all IntelliJ-derived IDEs have a setting to automatically detect what's being use. Is there anything we could do for Xcode ?

Reply:
Pray? :( 🙏 

I also think the default setting is spaces so that could cause even more issues potentially.

Reply:
I'm trying to think if there's any good way to support either method but I don't have enough experience here TBH